### ISSUE TITLE
Support for JSON-based indexes (json_extract)

### ISSUE THREAD

**Issue by michaeladler** [ariga/atlas#1436]
It would be great if atlas could also generate the schema for JSON-valued indexes. For example, using [sqlite](https://www.sqlite.org/json1.html):

```sql
CREATE TABLE foo (data json);
CREATE INDEX idx_data_id ON foo(json_extract(data, '$.id'));
```

Generate schema:

```bash
$ atlas schema inspect -u "sqlite://file:/git/example.db"
table "foo" {
  schema = schema.main
  column "data" {
    null = true
    type = json
  }
  index "idx_data_id" {
    on {
      expr = "<unsupported>"
    }
  }
}
schema "main" {
}
```

**Comment by a8m** [comment]
Thanks for reporting this, @michaeladler.
The SQLite driver does not support functional indexes (yet), but we definitely should support this. I'll address it soon and update this issue.

**Comment by a8m** [comment]
This issue was solved with #1447, and I released new binaries. Please, update Atlas and let me know if there are any other problems with it. Thanks 🙏

**Comment by michaeladler** [comment]
Thanks a lot for the quick response! It works for me :+1:
